lib/sender.js

```diff
@@ -85,7 +85,8 @@ class Sender {
       offset = 6;
     }
 
-    let payloadLength = data.length;
+    const dataLength = Buffer.byteLength(data);
```

Comment:
I'm not sure I understand, everything is already a buffer here.

Reply:
No it's not, that's the point. See https://github.com/websockets/ws/pull/1998/files#diff-ee0f1feef959a4232309df5db73baad8cd2b6d2d5be97f3ef6565a29093c61a8R353

Reply:
Correct I did not read the `send()` changes below. Anyway I think there is an implicit conversion to `Buffer` when writing to the socket?

Reply:
Yes, however that happens in native code and multiple chunks are converted at the same time. Less allocations and better perf.

Reply:
Hmm, unconvinced, `writevGeneric()` was  optmized for the "all buffers" case. Do you have benchmarks?

Reply:
Also I think this crashes if the user tries to write an `Array` or `Object`.

Reply:
https://github.com/websockets/ws/pull/1998#issuecomment-1005582961

Reply:
> Also, I think this crashes if the user tries to write an Array or Object.

Should that be possible? Also I think it already crashes if you do that. How does the PR change the behavior?

Reply:
```js
Buffer.byteLength([1,2,3]);
Buffer.byteLength(new String('foo'));
```

throws.

Reply:
Fixed